client/python/rappor.py

```diff
@@ -119,39 +119,37 @@ def from_csv(f):
     return p
 
 
-class _SimpleRandom(object):
+class _SecureRandom(object):
   """Returns an integer where each bit has probability p of being 1."""
 
-  def __init__(self, prob_one, num_bits, _rand=None):
+  def __init__(self, prob_one, num_bits):
     self.prob_one = prob_one
     self.num_bits = num_bits
-    self._rand = _rand or random.Random()
 
   def __call__(self):
     p = self.prob_one
-    rand_fn = self._rand.random  # cache it for speed
+    bytes = os.urandom(self.num_bits)
 
     r = 0
-    for i in xrange(self.num_bits):
-      bit = rand_fn() < p
+    for i, b in enumerate(xrange(self.num_bits)):
+      bit = b < p * 255.0
```

Comment:
Two issues:
1. The current version must be corrected to p * 256.0 (consider the case of p = 1.0)
2. I preferred the previous version, where randomness was sampled as a floating-point number. That way, p is not rounded to a multiple of 255 (or 256). For example, in the current code if p = .001, the output is going to be always 0.